src/main/webapp/script.js

```diff
@@ -173,6 +173,46 @@ function getGraphDisplay(graphNodes, graphEdges) {
     const node = evt.target;
     node.tip.show();
   });
+
+  const searchElement = document.getElementById('search');
+  searchElement.onchange = function() {
```

Comment:
Why not use a button for this? If the graph was very large, performing the operation could cause a noticeable delay.